### ISSUE TITLE
[Dropdown] shouldn’t be firing onChange during the keyboard navigation

### ISSUE THREAD

**Issue by batata004** [Semantic-Org/Semantic-UI#5055]
Hi,

I have a drop down element ("ui fluid search selection dropdown large elemento_area") in my page and if the user hits the letter "C" (for example) the dropdown shows a list of possible results that the user can choose. Ok, it works fine and it's really great. When the user selects some option I use the code below to do my stuff:

```
                $(".elemento_area").dropdown({
                
                    action:"hide",
                    onChange:function(value,text,$selectedItem) {
                    
                        window.location.href = "index.php?area=" + value;
                    
                    }
                
                });
```

Ok, it works fine too. The problem is that if the user types "C" and press one of the arrows of the keyboard (down/up) to navigate thgouth the options, as soon as the user presses down key, the event ONCHANGE is fired. This shouldnt happen cause the user still didnt decided what option he wants to choose, he is just using the arrow keys to navigate till the desired option (think about a lazey person or a handcapted).

I think there should be an event that fires only when user selects indeed the option he wants like "onselect". And I think the behaviour of "onchange" should make it's name change to "onnavigate"... 

Am I doing something wrong and this "onselect" event already exists? In the docs I couldnt find anything about this, only a handful of events.

**Comment by Banandrew** [comment]
It’s getting pretty ridiculous if you try to change the language with a keyboard on SUI’s documentation website, I think we need to tackle it ASAP.

**Comment by batata004** [comment]
My english is bad but I didnt understand what you said, are you blaming me for something I may have done wrong?

**Comment by Banandrew** [comment]
@batata004 Not at all! I described another test case and said that we need to fix this soon. 🍻

**Comment by batata004** [comment]
@Banandrew Ah ok, sorry. I thought you had said I changed the language in some wiki page of SUI's documentation. Now I got it, I hope someone can provide a fix to this and to the other serious bug I already reported many months ago related to modal dialogs which causes lots of problems and so far no fix was provided.

**Comment by Banandrew** [comment]
I’ve just stumbled upon this thread and realized that I mislabeled it—`selectOnKeydown` property does exactly that.

**Comment by batata004** [comment]
WORKED PERFECTLY!!!! I was waiting since February for this solution! This website https://www.deobraemobra.com.br relies heavily in semantic ui and I had to use a terrible hack in the search box at the top! Now I can easily press KEY DOWN in my keyboard to navigate inside the options and the onchange event does not fire.

@Banandrew great help of yours, again you surprised me with good news ;) This webiste is working perfectly now, I just removed the hack and it was pretty forward to make it work, just added "selectOnKeydown:false" and it worked!
